components/Buttons/src/MDCFloatingButton.m

```diff
@@ -71,55 +112,171 @@ - (instancetype)initWithFrame:(CGRect)frame shape:(MDCFloatingButtonShape)shape
 - (instancetype)initWithCoder:(NSCoder *)aDecoder {
   self = [super initWithCoder:aDecoder];
   if (self) {
+    [self commonMDCFloatingButtonInit];
     _shape = [aDecoder decodeIntegerForKey:MDCFloatingButtonShapeKey];
     // Required to migrate any previously-archived FloatingButtons from .largeIcon shape value
     if (@(_shape).integerValue >= 2) {
       _shape = MDCFloatingButtonShapeDefault;
     }
+    _mode = [aDecoder decodeIntegerForKey:MDCFloatingButtonModeKey];
+    _imageLocation = [aDecoder decodeIntegerForKey:MDCFloatingButtonImageLocationKey];
+    _imageTitleSpacing =
```

Comment:
Consider checking for the existence of the key before decoding _ivars.  Especially any with a default value != 0.

Reply:
Won't the keys always exist?

Reply:
Not from Interface Builder, they wouldn't. And in that case the defaults would be lost. Good catch!

Reply:
Ah I didn't think of that.

Reply:
Done